checkov/terraform/plan_parser.py

```diff
@@ -213,6 +217,22 @@ def _prepare_resource_block(
     return resource_block, block_type, prepared
 
 
+def _eval_after_unknown(changes: dict[str, Any], resource_conf: dict[str, Any]) -> None:
+    after_unknown = changes.get("change", {}).get(TF_PLAN_RESOURCE_AFTER_UNKNOWN)
+    if os.getenv('EVAL_TF_PLAN_AFTER_UNKNOWN') and after_unknown and isinstance(after_unknown, dict):
+        for k, v in after_unknown.items():
+            # We check if the value of the field is True. That would mean its value is known after the apply
+            # We also check whether the field is not already present in the conf since we do not want to
+            # override it. Overriding can actually cause losing its value
+            if v is True and k not in resource_conf:
+                # We set the value to 'true_after_unknown' and not its original value
+                # We need to set a constant other than a boolean (True/"true"),
+                # so it will not collide with actual possible values of those attributes
+                # In these cases, policies checking the existence of a value will succeed,
+                # but policies checking for concrete values will fail
+                resource_conf[k] = _clean_simple_type_list([TRUE_AFTER_UNKNOWN])
```

Comment:
if we always get `[TRUE_AFTER_UNKNOWN]` why do you need this function _clean_simple_type_list?
and why do we need this to be a list?

Reply:
Because this is how we build the conf in the plan_parser: `https://github.com/bridgecrewio/checkov/blob/57a8292d8dd5cdc87d4b9fd341f75843733fe681/checkov/terraform/plan_parser.py#L117`
In case the logic of the resource building changes, it will be reflected in this case as well. The list is just a wrapper since it fails otherwise in the parsing.